File: Iterator/python/BrowseHistory.py
from StringIterator import StringIterator


class BrowseHistory:
    __urls = []

    def getUrls(self) -> list:
        return self.__urls

    def createIterator(self) -> StringIterator:
        return StringIterator(self)

    def push(self, url: str) -> None:
        self.__urls.append(url)

    def pop(self) -> str:
        return self.__urls.pop()

File: Iterator/dart/BrowseHistory.dart
import 'Iterator.dart';
import 'StringIterator.dart';

class BrowseHistory {
  List<String> _urls = [];

  List<String> get urls => _urls;

  Iterator createIterator() {
    return new StringIterator(history: this);
  }

  void push(String url) {
    urls.add(url);
  }

  String pop() {
    return urls.removeLast();
  }
}
